components/esp_websocket_client/esp_websocket_client.c

```diff
@@ -519,6 +522,10 @@ static esp_err_t esp_websocket_client_create_transport(esp_websocket_client_hand
             } else {
                 esp_transport_ssl_set_client_key_data_der(ssl, client->config->client_key, client->config->client_key_len);
             }
+#if CONFIG_ESP_TLS_USE_DS_PERIPHERAL
+        } else if (client->config->client_ds_data) {
+            esp_transport_ssl_set_ds_data(ssl, client->config->client_ds_data);
```

Comment:
The signature here is:

```c
void esp_transport_ssl_set_ds_data(esp_transport_handle_t t, void *ds_data)
```

Hence I'm using `void *` and not `esp_ds_data_ctx_t *` to avoid a dependency on Mbed TLS.